### PR TITLE
WIP: Remove an object allocation from instrumentation decorators which interact with fields.

### PR DESCRIPTION
**Pull request by treverj**
See discussion here for more info, this is intended as a POC

### CODE REVIEW

File: src/main/java/graphql/execution/instrumentation/dataloader/FieldLevelTrackingApproach.java [hunk 1]
```diff
@@ -29,60 +32,63 @@
...
-        private final LevelMap fetchCountPerLevel = new LevelMap();
-        private final LevelMap expectedStrategyCallsPerLevel = new LevelMap();
-        private final LevelMap happenedStrategyCallsPerLevel = new LevelMap();
-        private final LevelMap happenedOnFieldValueCallsPerLevel = new LevelMap();
+        private final Map<Integer, Integer> expectedFetchCountPerLevel = new LinkedHashMap<>();
+        private final Map<Integer, Integer> fetchCountPerLevel = new LinkedHashMap<>();
+        private final Map<Integer, Integer> expectedStrategyCallsPerLevel = new LinkedHashMap<>();
+        private final Map<Integer, Integer> happenedStrategyCallsPerLevel = new LinkedHashMap<>();
+        private final Map<Integer, Integer> happenedOnFieldValueCallsPerLevel = new LinkedHashMap<>();
+
```
Comment (bbakerman) [inline src/main/java/graphql/execution/instrumentation/dataloader/FieldLevelTrackingApproach.java:40]: Why did you change this.  Our custom LevelMap (array of ints) surely is more memory efficeient than a LinkedHashMap

File: src/main/java/graphql/execution/ExecutionStrategy.java [hunk 2]
```diff
@@ -692,7 +692,7 @@ private Object handleCoercionProblem(ExecutionContext context, ExecutionStrategy
     }
 
     protected GraphQLObjectType resolveType(ExecutionContext executionContext, ExecutionStrategyParameters parameters, GraphQLType fieldType) {
-        return resolvedType.resolveType(executionContext, parameters.getField(), parameters.getSource(), parameters.getExecutionStepInfo(), fieldType, parameters.getLocalContext());
+        return resolvedType.resolveType(executionContext, parameters.getField(), parameters.getSource(), parameters.getExecutionStepInfo().getArguments(), fieldType);
```
Comment (bbakerman) [inline src/main/java/graphql/execution/ExecutionStrategy.java:695]: why these changes?  What has this got to do with instrumentation state?

File: src/main/java/graphql/execution/instrumentation/dataloader/FieldLevelTrackingApproach.java (hunk 1 above)
Comment (treverj) [reply src/main/java/graphql/execution/instrumentation/dataloader/FieldLevelTrackingApproach.java:40]: Apologies, we forked the repo internally to analyse and test these changes and when I have tried to move this commit out into the public repo for you to see I have somehow pulled in some unrelated things (that to be honest I haven't seen before), let me fix this PR.

File: src/main/java/graphql/execution/ExecutionStrategy.java (hunk 2 above)
Comment (treverj) [reply src/main/java/graphql/execution/ExecutionStrategy.java:695]: same as above

**Comment by treverj** [comment]
Apologies, this PR has now been fixed to just include the correct changes. Sorry for the confusion.

**Comment by bbakerman** [comment]
This has been addressed in another PR

**Pull request closed without merging**